src/screens/OrgPost/Posts.tsx

```diff
@@ -56,8 +56,24 @@ const PostsRenderer: React.FC<InterfacePostsRenderer> = ({
   if (loading) return <Loader />;
   if (error) return <div>Error loading posts</div>;
 
-  // Rest of the component remains the same...
-  // (Previous implementation of createAttachments, renderPostCard, and rendering logic)
+  // The generateUniqueObjectName function properly handles both image and video URLs
+  const generateUniqueObjectName = (
+    postId: string,
+    fileUrl: string,
+    fileType: string
+  ): string => {
+    // Extract file extension from URL or use default based on type
+    const extension = fileUrl.split('.').pop()?.toLowerCase() || 
```

Comment:
1. The develop-postgres branch of the API no longer uses `fileurl` therefore your mock is incorrect. Using `fileurl` in the app will cause it to malfunction.
2. You need to use `objectName` instead 
3. You also cannot use `objectName` in the preview. This is impossible. You have to use the  browser's URL object method to preview the images and the MinioUpload.ts helper function for uploading images.